lib/raygun.breadcrumbs.ts

```diff
@@ -151,3 +155,36 @@ export function clear() {
   );
   asyncLocalStorage.enterWith([]);
 }
+
+export function setupConsoleBreadcrumbs() {
+  // Map console methods to a Breadcrumb level
+  const consoleMethods: [keyof typeof console, Breadcrumb["level"]][] = [
+    ["debug", "debug"],
+    ["log", "info"],
+    ["info", "info"],
+    ["warn", "warning"],
+    ["error", "error"],
+  ];
+
+  // Extend all console methods to call to addBreadcrumb
```

Comment:
I haven't run this PR yet, but what happens with the internal (as in: internal to our provider) calls of `console.<something>` ? Wouldn't those trigger this logic as well?

Reply:
If the user configured `setupConsoleBreadcrumbs` to `true` then yes, all calls to `console.log` get registered as breadcrumbs as well, not just the ones the user does, but _all of them_ including the ones in 3rd party packages.

Reply:
>If the user configured setupConsoleBreadcrumbs to true then yes, all calls to console.log get registered as breadcrumbs as well, not just the ones the user does, but all of them including the ones in 3rd party packages.

Hm, when I suggested trying that in the old PR comment I wasn't really clear of this side effect. I'm not sure that this is such a good feature then. It seems it'll lead to a lot of stuff being logged as breadcrumbs that users would quite possibly not want to be logged in there. Very noisy and cluttered as a result.

Should we pull this PR from 1.2.0 and have a further though on the feature itself?

@sumitramanga Any opinions on this?

Reply:
Yeah I agree with Kai. I can check out what the other providers are doing and see if this is the correct behaviour to have

Reply:
Seems that the `raygun4js` also does it: https://github.com/MindscapeHQ/raygun4js/blob/2d0eeb333b1c3e28df26c9597e78b034b30afb91/src/raygun.breadcrumbs.js#L207

I would be surprised if any other implementation besides JavaScript does it, since it is fairly hacky to override the original `console.log` methods, and I am not sure other platforms may support it.

Otherwise, the feature is disabled by default, so it won't affect users unless they enable it.

Reply:
I acknowledge all of your points - it's disabled by default and it's the same in RG4JS.

But: I'm not sure that - given the side-effects when used - this is actually a good feature in the first place. I mean, sure we can have it and it's implemented as well as it can be done, but the feature feels a bit meh now and almost not good practice to even offer it.